### ISSUE TITLE
There seems to be an infinite loop in `infra/helper.py check_build`

### ISSUE THREAD

**Issue by evverx** [google/oss-fuzz#1405]
Trying to reproduce https://github.com/google/oss-fuzz/issues/1330, I noticed that `check_build` got stuck. It turns out that when [`jobs -p`](http://git.savannah.gnu.org/cgit/bash.git/tree/jobs.c?id=7de27456f6494f5f9c11ea1c19024d0024f31112#n1808) is used in a script, `bash` doesn't seem to mark jobs as ["J_NOTIFIED"](http://git.savannah.gnu.org/cgit/bash.git/tree/jobs.c?id=7de27456f6494f5f9c11ea1c19024d0024f31112#n1833), which leads to an infinite loop as soon as the number of jobs that has been run equals the number returned by `nproc`. It can be reproduced by applying the following patch:
```diff
diff --git a/infra/base-images/base-runner/test_all b/infra/base-images/base-runner/test_all
index 6a071c1..f5482d6 100755
--- a/infra/base-images/base-runner/test_all
+++ b/infra/base-images/base-runner/test_all
@@ -22,7 +22,7 @@ ALLOWED_BROKEN_TARGETS_PERCENTAGE=10
 TOTAL_TARGETS_COUNT=0

 # Number of CPUs available, this is needed for running tests in parallel.
-NPROC=$(nproc)
+NPROC=1

 # Directories where bad build check results will be written to.
 VALID_TARGETS_DIR="/tmp/valid_fuzz_targets"
```
and can be fixed by using `jobs -rp` instead of `jobs -p`:
```diff
diff --git a/infra/base-images/base-runner/test_all b/infra/base-images/base-runner/test_all
index 6a071c1..c9b6f5e 100755
--- a/infra/base-images/base-runner/test_all
+++ b/infra/base-images/base-runner/test_all
@@ -57,10 +57,10 @@ for FUZZER_BINARY in $(find $OUT/ -executable -type f); do
   TOTAL_TARGETS_COUNT=$[$TOTAL_TARGETS_COUNT+1]

   # Do not spawn more processes than the number of CPUs available.
-  n_child_proc=$(jobs -p | wc -l)
+  n_child_proc=$(jobs -rp | wc -l)
   while [ "$n_child_proc" -eq "$NPROC" ]; do
     sleep 4
-    n_child_proc=$(jobs -p | wc -l)
+    n_child_proc=$(jobs -rp | wc -l)
   done
 done
```
I'd be happy to create a PR, but I can't seem to use my email address to sign the CLA.

**Comment by evverx** [comment]
So as not to forget what I was talking about, I'll leave a reduced version of `test_all` showing the infinite loop in action:
```sh
#!/bin/bash -u

NPROC=1

sleep 4 &

n_child_proc=$(jobs -p | wc -l)
while [ "$n_child_proc" -eq "$NPROC" ]; do
  sleep 4
  n_child_proc=$(jobs -p | wc -l)
done
```

**Comment by inferno-chromium** [comment]
Assigning to current sheriff, if patch looks ok, please commit.

**Comment by Dor1s** [comment]
Interesting! Thanks for the report. I guess it's happening only with `NPROC=1`, because otherwise we leave the loop and then do `wait` which should work fine

**Comment by Dor1s** [comment]
Jonathan has landed the fix (#1411), thanks everyone!
